homeassistant/components/sensor/foobot.py

```diff
@@ -0,0 +1,147 @@
+"""
+Support for the Foobot indoor air quality monitor.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/sensor.foobot/
+"""
+import asyncio
+import logging
+from datetime import timedelta, datetime
+from collections import OrderedDict
+
+import aiohttp
+import voluptuous as vol
+
+from homeassistant.exceptions import PlatformNotReady
+import homeassistant.helpers.config_validation as cv
+from homeassistant.const import (
+    ATTR_TIME, ATTR_TEMPERATURE, CONF_TOKEN, CONF_USERNAME, TEMP_CELSIUS)
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+from homeassistant.helpers.config_validation import PLATFORM_SCHEMA
+from homeassistant.helpers.entity import Entity
+
+REQUIREMENTS = ['foobot_async==0.3.0']
+
+_LOGGER = logging.getLogger(__name__)
+
+ATTR_HUMIDITY = 'humidity'
+ATTR_PM2_5 = 'PM2.5'
+ATTR_CARBON_DIOXIDE = 'CO2'
+ATTR_VOLATILE_ORGANIC_COMPOUNDS = 'VOC'
+ATTR_FOOBOT_INDEX = 'index'
+
+SENSOR_TYPES = OrderedDict([('time', [ATTR_TIME, 's']),
+                            ('pm', [ATTR_PM2_5, 'µg/m3', 'mdi:cloud']),
+                            ('tmp', [ATTR_TEMPERATURE, TEMP_CELSIUS,
+                                     'mdi:thermometer']),
+                            ('hum', [ATTR_HUMIDITY, '%', 'mdi:water-percent']),
+                            ('co2', [ATTR_CARBON_DIOXIDE, 'ppm',
+                                     'mdi:periodic-table-co2']),
+                            ('voc', [ATTR_VOLATILE_ORGANIC_COMPOUNDS, 'ppb',
+                                     'mdi:cloud']),
+                            ('allpollu', [ATTR_FOOBOT_INDEX, '%',
+                                          'mdi:percent'])])
+
+SCAN_INTERVAL = timedelta(minutes=10)
+PARALLEL_UPDATES = 1
+
+TIMEOUT = 10
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_TOKEN): cv.string,
+    vol.Required(CONF_USERNAME): cv.string,
+})
+
+
+@asyncio.coroutine
+def async_setup_platform(hass, config, async_add_devices, discovery_info=None):
+    """Set up the devices associated with the account."""
+    from foobot_async import FoobotClient
+
+    token = config.get(CONF_TOKEN)
+    username = config.get(CONF_USERNAME)
+
+    client = FoobotClient(token, username,
+                          async_get_clientsession(hass),
+                          timeout=TIMEOUT)
+    data = {}
+    dev = []
+    try:
+        devices = yield from client.get_devices()
+        _LOGGER.debug("The following devices were found: %s", devices)
+        for device in devices:
+            for sensor_type in SENSOR_TYPES.keys():
+                if sensor_type != 'time':
+                    foobot_sensor = FoobotSensor(client, data, device,
+                                                 sensor_type)
+                    dev.append(foobot_sensor)
+    except (aiohttp.client_exceptions.ClientConnectorError,
+            asyncio.TimeoutError, client.ClientError):
```

Comment:
`client.ClientError`: Did you mean `FoobotClient.ClientError`? (I could be wrong about this, just making sure)

Reply:
Indeed, good catch.